Alignment/APEEstimation/plugins/ApeTreeCreateDefault.cc

```diff
@@ -0,0 +1,473 @@
+    ReducedTrackerTreeVariables tkTreeVar;
+    
+    //Loop over all Modules
+    for(int module = 0; module < nModules; ++module){
+      tkTree->GetEntry(module);
+      
+      if(sectorCounter==1){
+        tkTreeVar.subdetId = subdetId;
+        tkTreeVar.nStrips = nStrips;
+        tkTreeVar.uDirection = uDirection;
+        tkTreeVar.vDirection = vDirection;
+        tkTreeVar.wDirection = wDirection;
+        m_tkTreeVar_[rawId] = tkTreeVar;
+      }
+      //Check if modules from Sector builder equal those from TrackerTree
+      if(!this->checkModuleIds(rawId,v_rawId))continue;
+      if(!this->checkModuleIds(subdetId,v_subdetId))continue;
+      if(!this->checkModuleIds(layer,v_layer))continue;
+      if(!this->checkModuleIds(side,v_side))continue;
+      if(!this->checkModuleIds(half,v_half))continue;
+      if(!this->checkModuleIds(rod,v_rod))continue;
+      if(!this->checkModuleIds(ring,v_ring))continue;
+      if(!this->checkModuleIds(petal,v_petal))continue;
+      if(!this->checkModuleIds(blade,v_blade))continue;
+      if(!this->checkModuleIds(panel,v_panel))continue;
+      if(!this->checkModuleIds(outerInner,v_outerInner))continue;
+      if(!this->checkModuleIds(module,v_module))continue;
+      if(!this->checkModuleIds(nStrips,v_nStrips))continue;
+      if(!this->checkModuleBools(isDoubleSide,v_isDoubleSide))continue;
+      if(!this->checkModuleBools(isRPhi,v_isRPhi))continue;
+      if(!this->checkModuleBools(isStereo,v_isStereo))continue;
+      if(!this->checkModuleDirections(uDirection,v_uDirection))continue;
+      if(!this->checkModuleDirections(vDirection,v_vDirection))continue;
+      if(!this->checkModuleDirections(wDirection,v_wDirection))continue;
+      if(!this->checkModulePositions(posR,v_posR))continue;
+      if(!this->checkModulePositions(posPhi,v_posPhi))continue;
+      if(!this->checkModulePositions(posEta,v_posEta))continue;
+      if(!this->checkModulePositions(posX,v_posX))continue;
+      if(!this->checkModulePositions(posY,v_posY))continue;
+      if(!this->checkModulePositions(posZ,v_posZ))continue;
+      
+      tkSector.v_rawId.push_back(rawId);
+      bool moduleSelected(false);
+      for(auto const & i_rawId : allSectors.v_rawId){
+        if(rawId == i_rawId)moduleSelected = true;
+      }
+      if(!moduleSelected)allSectors.v_rawId.push_back(rawId);
+    }
+    
+    // Stops you from combining pixel and strip detector into one sector
+    bool isPixel(false);
+    bool isStrip(false);
+    for(auto const & i_rawId : tkSector.v_rawId){
+      if(m_tkTreeVar_[i_rawId].subdetId==PixelSubdetector::PixelBarrel || m_tkTreeVar_[i_rawId].subdetId==PixelSubdetector::PixelEndcap){
+        isPixel = true;
+      }
+      if(m_tkTreeVar_[i_rawId].subdetId==StripSubdetector::TIB || m_tkTreeVar_[i_rawId].subdetId==StripSubdetector::TOB ||
+         m_tkTreeVar_[i_rawId].subdetId==StripSubdetector::TID || m_tkTreeVar_[i_rawId].subdetId==StripSubdetector::TEC){
+        isStrip = true;
+      }
```

Comment:
you could condense these two if statements like this:
```
switch (m_tkTreeVar_[i_rawId].subdetId) {
case PixelSubdetector::PixelBarrel:
case PixelSubdetector::PixelEncap:
  isPixel = true;
  break;
case StripSubdetector::TIB:
case StripSubdetector::TOB:
case StripSubdetector::TID:
case StripSubdetector::TEC:
  isStrip = true;
  break
}
```
I believe this is more readable and likely also more efficient, as you check `m_tkTreeVar_[i_rawId].subdetId` only once.